server/service/src/pricing/item_price.rs

```diff
@@ -50,19 +50,36 @@ pub fn get_pricing_for_item(
         None => None, // No default price list found, no price
     };
 
-    // 2. Lookup the discount list
-    // Find the first discount list that has the item (not trying to be clever here, just using the first one found)
-    let discount_master_list = MasterListRepository::new(&ctx.connection)
-        .query_by_filter(
-            MasterListFilter::new()
-                .is_discount_list(true)
-                .item_id(EqualFilter::equal_to(&input.item_id)),
-        )?
-        .pop();
+    // 2. Check if we have a name, and that name is not a patient
+    let is_patient = match &input.customer_name_id {
+        Some(customer_name_id) => {
+            let num_patients = PatientRepository::new(&ctx.connection).count(
```

Comment:
There is `query_one` on PatientRepository and there is also `is_some()` on Option

Reply:
I thought this was arguably more efficient for the database, happy either way though...